sklearn/df_clsf.py

```diff
@@ -54,25 +57,49 @@
            'num_trees', 'n_classes', 'accuracy', 'time')
 params.n_classes = len(np.unique(y))
 params.size = size_str(X.shape)
-params.dtype = X.dtype
 
-print_header(columns, params)
-
-# Time fit and predict
 fit_time, _ = time_mean_min(clf.fit, X, y,
                             outer_loops=params.fit_outer_loops,
                             inner_loops=params.fit_inner_loops,
                             goal_outer_loops=params.fit_goal,
                             time_limit=params.fit_time_limit,
                             verbose=params.verbose)
-print_row(columns, params, function='df_clsf.fit', time=fit_time)
-
 predict_time, y_pred = time_mean_min(clf.predict, X,
                                      outer_loops=params.predict_outer_loops,
                                      inner_loops=params.predict_inner_loops,
                                      goal_outer_loops=params.predict_goal,
                                      time_limit=params.predict_time_limit,
                                      verbose=params.verbose)
 acc = 100 * accuracy_score(y_pred, y)
-print_row(columns, params, function='df_clsf.predict', time=predict_time,
+
+if params.output_format == "csv":
+    print_header(columns, params)
```

Comment:
Maybe it would be possible to factor out this stuff into something in `bench.py`? Maybe results can be packed into some sort of dictionary, maybe the same dictionary used for the json result, and that could then be printed in the expected csv format. The only reason to separate `print_header` and `print_row` originally was to allow for real-time feedback from the benchmark that it had finished the timing of part of the benchmark. If we abandon that entirely, they can also be changed into one function call. For example, `params.size` could turn into `params.shape` and contain the actual shape of the input, not a string representation.

If it is required to rename e.g. `df_clsf` to `decision_forest_classification`, that should also be done for CSV output.

Reply:
CSV/JSON outputs were merged in one function and moved to bench.py. I think it will be enough for this pull request.

Reply:
Looks good, thanks! Benchmark files have now regained their simplicity.